DeepSpeech.ipynb

```diff
@@ -1320,170 +1292,390 @@
+    "    \n",
+    "    # Requirements to display a WER report\n",
+    "    if query_report:\n",
+    "        # Reset accuracy\n",
+    "        total_accuracy = 0.0\n",
+    "        # Create report results tuple\n",
+    "        report_results = ([],[],[],[])\n",
+    "        # Extend the session.run parameters\n",
+    "        params.append([results_params, avg_accuracy])\n",
     "\n",
-    "    # Create a saver to checkpoint the model\n",
-    "    saver = tf.train.Saver(tf.all_variables())\n",
+    "    # Loop over the batches\n",
+    "    for batch in range(int(batches_per_device)):\n",
+    "        extra_params = { }\n",
+    "        if do_training and do_fulltrace:\n",
+    "            loss_run_metadata            = tf.RunMetadata()\n",
+    "            extra_params['options']      = tf.RunOptions(trace_level=tf.RunOptions.FULL_TRACE)\n",
+    "            extra_params['run_metadata'] = loss_run_metadata\n",
+    "\n",
+    "        # Compute the batch\n",
+    "        result = session.run(params, **extra_params)\n",
+    "        \n",
+    "        # Offset to read from result list\n",
+    "        offset = 0\n",
+    "        \n",
+    "        # Add batch to loss\n",
+    "        total_loss += result[offset]\n",
+    "        offset = offset + 1\n",
+    "        \n",
+    "        if do_training:\n",
+    "            # Log all variable states in current step\n",
+    "            step = epoch * data_set.total_batches + batch * len(available_devices)\n",
+    "            writer.add_summary(result[offset][1], step)\n",
+    "            if do_fulltrace:\n",
+    "                writer.add_run_metadata(loss_run_metadata, 'loss_epoch%d_batch%d'   % (epoch, batch))\n",
+    "            writer.flush()\n",
+    "            offset = offset + 1\n",
     "\n",
-    "    # Init all variables in session\n",
-    "    session.run(tf.initialize_all_variables())\n",
+    "        if query_report:\n",
+    "            # Collect individual sample results\n",
+    "            collect_results(report_results, result[offset][0])\n",
+    "            # Add batch to total_accuracy\n",
+    "            total_accuracy += result[offset][1]\n",
+    "            \n",
+    "    # Returning the batch set result tuple\n",
+    "    loss = total_loss / batches_per_device\n",
+    "    if query_report:\n",
+    "        return (loss, total_accuracy / batches_per_device, report_results)\n",
+    "    else:\n",
+    "        return (loss, None, None)\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The following routine will print a report from a provided batch set result tuple. It takes a caption for titling the output plus the batch set result tuple. If the batch set result tuple contains accuracy and a report results tuple, a complete WER report will be calculated, printed and its mean WER returned. Otherwise it will just print the loss and return `None`."
```

Comment:
@Cwiiis I'm going to ignore this for now as it's Friday night and I want to get this merged and the run going.